buildtools/dep/dep.go

```diff
@@ -22,39 +37,62 @@ type Project struct {
 	Version  string
 }
 
-// A Lockfile contains the contents of a dep lockfile. Lockfiles are resolvers.
-type Lockfile struct {
-	Projects []Project
+// Resolve determines if the Gopkg.toml dep file has any rules that apply
+func (r Resolver) Resolve(importpath string) (pkg.Import, error) {
+	if r.Manifest.IsIgnored(importpath) {
+		return pkg.Import{}, buildtools.ErrPackageIsIgnored
+	}
 
-	normalized map[string]pkg.Import // A normalized map of package import paths to revisions.
+	return r.Lockfile.Resolve(importpath)
+}
+
+// IsIgnored finds packages to ignore while handling the possibility of a wildcard
```

Comment:
[nit] Here and elsewhere, declaration comments should always be complete sentences (with punctuation).